From b604dd17d70382d91dcfeeb738dfd0149e69d04e Mon Sep 17 00:00:00 2001
From: Jakub Jirutka <jakub@jirutka.cz>
Date: Mon, 27 Sep 2021 21:16:03 +0200
Subject: [PATCH 2/3] Remove unused/surplus xcb-related includes
Patch-Source: https://github.com/lbonn/rofi/pull/31

Includes in the rofi code base are a complete mess; there are many
needless includes and many files rely on transitive includes.

I used include-what-you-use tool to find them. There are much more,
I cherry picked only suggestions relevant to the goal - allow building
rofi without XCB dependencies.

This doesn't break X11 support and it can be upstreamed (after rebasing
and removing changes in wayland-stuff).
---
 include/widgets/textbox.h           | 1 -
 include/widgets/widget.h            | 2 --
 include/xrmoptions.h                | 1 -
 source/dialogs/drun.c               | 1 -
 source/display.c                    | 1 -
 source/helper.c                     | 1 -
 source/rofi-icon-fetcher.c          | 2 --
 source/rofi.c                       | 3 ---
 source/wayland/view.c               | 5 -----
 source/widgets/scrollbar.c          | 1 -
 source/widgets/textbox.c            | 1 -
 source/xrmoptions.c                 | 4 ----
 test/helper-config-cmdline-parser.c | 3 ---
 test/helper-expand.c                | 3 ---
 test/helper-pidfile.c               | 3 ---
 test/helper-test.c                  | 3 ---
 test/helper-tokenize.c              | 3 ---
 test/mode-test.c                    | 2 --
 test/textbox-test.c                 | 4 +---
 test/theme-parser-test.c            | 3 ---
 test/widget-test.c                  | 1 -
 21 files changed, 1 insertion(+), 47 deletions(-)

diff --git a/include/widgets/textbox.h b/include/widgets/textbox.h
index 3386557c..237b1dd1 100644
--- a/include/widgets/textbox.h
+++ b/include/widgets/textbox.h
@@ -35,7 +35,6 @@
 #include <pango/pango-fontmap.h>
 #include <pango/pango.h>
 #include <pango/pangocairo.h>
-#include <xkbcommon/xkbcommon.h>
 
 /**
  * @defgroup Textbox Textbox
diff --git a/include/widgets/widget.h b/include/widgets/widget.h
index c3aac979..3c349272 100644
--- a/include/widgets/widget.h
+++ b/include/widgets/widget.h
@@ -30,8 +30,6 @@
 #include "keyb.h"
 #include <cairo.h>
 #include <glib.h>
-#include <xcb/xcb.h>
-#include <xcb/xproto.h>
 /**
  * @defgroup widget widget
  *
diff --git a/include/xrmoptions.h b/include/xrmoptions.h
index 4545ecb4..f8fda7fc 100644
--- a/include/xrmoptions.h
+++ b/include/xrmoptions.h
@@ -28,7 +28,6 @@
 #ifndef ROFI_XRMOPTIONS_H
 #define ROFI_XRMOPTIONS_H
 #include "theme.h"
-#include "xcb.h"
 // Big thanks to Sean Pringle for this code.
 
 /**
diff --git a/source/dialogs/drun.c b/source/dialogs/drun.c
index d79cf677..a6065156 100644
--- a/source/dialogs/drun.c
+++ b/source/dialogs/drun.c
@@ -53,7 +53,6 @@
 #include "settings.h"
 #include "timings.h"
 #include "widgets/textbox.h"
-#include "xcb.h"
 
 #include "rofi-icon-fetcher.h"
 
diff --git a/source/display.c b/source/display.c
index dd2ac9bd..ea167dda 100644
--- a/source/display.c
+++ b/source/display.c
@@ -1,6 +1,5 @@
 #include "keyb.h"
 #include <glib.h>
-#include <xcb/xkb.h>
 
 #include "display-internal.h"
 #include "display.h"
diff --git a/source/helper.c b/source/helper.c
index c549419d..84771b71 100644
--- a/source/helper.c
+++ b/source/helper.c
@@ -35,7 +35,6 @@
 #include "rofi.h"
 #include "settings.h"
 #include "view.h"
-#include "xcb.h"
 #include <ctype.h>
 #include <errno.h>
 #include <fcntl.h>
diff --git a/source/rofi-icon-fetcher.c b/source/rofi-icon-fetcher.c
index b0109e0f..37fe949f 100644
--- a/source/rofi-icon-fetcher.c
+++ b/source/rofi-icon-fetcher.c
@@ -30,7 +30,6 @@
 
 #include "config.h"
 #include <stdlib.h>
-#include <xcb/xproto.h>
 
 #include "helper.h"
 #include "rofi-icon-fetcher.h"
@@ -39,7 +38,6 @@
 
 #include "keyb.h"
 #include "view.h"
-#include "xcb.h"
 
 #include "nkutils-enum.h"
 #include "nkutils-xdg-theme.h"
diff --git a/source/rofi.c b/source/rofi.c
index f8526f80..ac9e9213 100644
--- a/source/rofi.c
+++ b/source/rofi.c
@@ -40,12 +40,9 @@
 #include <sysexits.h>
 #include <time.h>
 #include <unistd.h>
-#include <xcb/xcb.h>
 
 #include <glib-unix.h>
 
-#include <libgwater-xcb.h>
-
 #ifdef USE_NK_GIT_VERSION
 #include "nkutils-git-version.h"
 #ifdef NK_GIT_VERSION
diff --git a/source/wayland/view.c b/source/wayland/view.c
index 78e08f6b..85a8b010 100644
--- a/source/wayland/view.c
+++ b/source/wayland/view.c
@@ -41,11 +41,7 @@
 
 #include <cairo.h>
 
-/** Indicated we understand the startup notification api is not yet stable.*/
-#define SN_API_NOT_YET_FROZEN
 #include "rofi.h"
-#include <libsn/sn.h>
-
 #include "settings.h"
 #include "timings.h"
 
@@ -54,7 +50,6 @@
 #include "helper-theme.h"
 #include "helper.h"
 #include "mode.h"
-#include "xrmoptions.h"
 
 #include "view-internal.h"
 #include "view.h"
diff --git a/source/widgets/scrollbar.c b/source/widgets/scrollbar.c
index aaa23b68..db8b0df6 100644
--- a/source/widgets/scrollbar.c
+++ b/source/widgets/scrollbar.c
@@ -30,7 +30,6 @@
 #include "widgets/listview.h"
 #include "widgets/textbox.h"
 #include <glib.h>
-#include <xkbcommon/xkbcommon.h>
 
 #include "theme.h"
 
diff --git a/source/widgets/textbox.c b/source/widgets/textbox.c
index 57d76844..f9315644 100644
--- a/source/widgets/textbox.c
+++ b/source/widgets/textbox.c
@@ -36,7 +36,6 @@
 #include <glib.h>
 #include <math.h>
 #include <string.h>
-#include <xcb/xcb.h>
 
 #include "theme.h"
 
diff --git a/source/xrmoptions.c b/source/xrmoptions.c
index 5640a283..a1271066 100644
--- a/source/xrmoptions.c
+++ b/source/xrmoptions.c
@@ -32,16 +32,12 @@
 #include "rofi-types.h"
 #include "rofi.h"
 #include "settings.h"
-#include "xcb-internal.h"
-#include "xcb.h"
 #include <ctype.h>
 #include <glib.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
 #include <unistd.h>
-#include <xcb/xcb.h>
-#include <xcb/xkb.h>
 
 ThemeWidget *rofi_configuration = NULL;
 
diff --git a/test/helper-config-cmdline-parser.c b/test/helper-config-cmdline-parser.c
index f12e9079..d6624dbf 100644
--- a/test/helper-config-cmdline-parser.c
+++ b/test/helper-config-cmdline-parser.c
@@ -31,10 +31,7 @@
 #include <stdio.h>
 #include <helper.h>
 #include <string.h>
-#include <xcb/xcb_ewmh.h>
 #include "display.h"
-#include "xcb.h"
-#include "xcb-internal.h"
 #include "rofi.h"
 #include "settings.h"
 #include "widgets/textbox.h"
diff --git a/test/helper-expand.c b/test/helper-expand.c
index b673e190..9fc3cbf8 100644
--- a/test/helper-expand.c
+++ b/test/helper-expand.c
@@ -31,11 +31,8 @@
 #include <stdio.h>
 #include <helper.h>
 #include <string.h>
-#include <xcb/xcb_ewmh.h>
 #include "theme.h"
 #include "display.h"
-#include "xcb.h"
-#include "xcb-internal.h"
 #include "rofi.h"
 #include "settings.h"
 #include "widgets/textbox.h"
diff --git a/test/helper-pidfile.c b/test/helper-pidfile.c
index fabb6c4c..8558ab72 100644
--- a/test/helper-pidfile.c
+++ b/test/helper-pidfile.c
@@ -31,10 +31,7 @@
 #include <stdio.h>
 #include <helper.h>
 #include <string.h>
-#include <xcb/xcb_ewmh.h>
 #include "display.h"
-#include "xcb.h"
-#include "xcb-internal.h"
 #include "rofi.h"
 #include "settings.h"
 #include "widgets/textbox.h"
diff --git a/test/helper-test.c b/test/helper-test.c
index 71109fd4..3af19d44 100644
--- a/test/helper-test.c
+++ b/test/helper-test.c
@@ -31,11 +31,8 @@
 #include <stdio.h>
 #include <helper.h>
 #include <string.h>
-#include <xcb/xcb_ewmh.h>
 #include "theme.h"
 #include "display.h"
-#include "xcb.h"
-#include "xcb-internal.h"
 #include "rofi.h"
 #include "settings.h"
 #include "rofi-icon-fetcher.h"
diff --git a/test/helper-tokenize.c b/test/helper-tokenize.c
index c510cb49..2b05eec7 100644
--- a/test/helper-tokenize.c
+++ b/test/helper-tokenize.c
@@ -31,11 +31,8 @@
 #include <stdio.h>
 #include <helper.h>
 #include <string.h>
-#include <xcb/xcb_ewmh.h>
 #include "display.h"
 #include "theme.h"
-#include "xcb.h"
-#include "xcb-internal.h"
 #include "rofi.h"
 #include "settings.h"
 #include "rofi-types.h"
diff --git a/test/mode-test.c b/test/mode-test.c
index 2c4c7216..fdc0de15 100644
--- a/test/mode-test.c
+++ b/test/mode-test.c
@@ -36,11 +36,9 @@
 #include <mode.h>
 #include <mode-private.h>
 #include <dialogs/help-keys.h>
-#include <xkbcommon/xkbcommon.h>
 #include "theme.h"
 #include "rofi.h"
 #include "display.h"
-#include "xcb.h"
 #include "widgets/textbox.h"
 #include <keyb.h>
 #include <helper.h>
diff --git a/test/textbox-test.c b/test/textbox-test.c
index 9c6a7388..7b7127d2 100644
--- a/test/textbox-test.c
+++ b/test/textbox-test.c
@@ -33,12 +33,10 @@
 #include <glib.h>
 #include <history.h>
 #include <string.h>
-#include <xcb/xcb.h>
 #include <widgets/textbox.h>
 #include <rofi.h>
-#include <cairo-xlib.h>
+#include <cairo.h>
 #include "display.h"
-#include "xcb.h"
 #include "settings.h"
 #include "xrmoptions.h"
 
diff --git a/test/theme-parser-test.c b/test/theme-parser-test.c
index 2915c46c..ccae1bb9 100644
--- a/test/theme-parser-test.c
+++ b/test/theme-parser-test.c
@@ -31,12 +31,9 @@
 #include <stdio.h>
 #include <helper.h>
 #include <string.h>
-#include <xcb/xcb_ewmh.h>
-#include "xcb-internal.h"
 #include "rofi.h"
 #include "settings.h"
 #include "display.h"
-#include "xcb.h"
 #include "theme.h"
 #include "css-colors.h"
 #include "widgets/widget-internal.h"
diff --git a/test/widget-test.c b/test/widget-test.c
index ca8d41bf..15577033 100644
--- a/test/widget-test.c
+++ b/test/widget-test.c
@@ -37,7 +37,6 @@
 #include "rofi.h"
 #include "display.h"
 #include "xrmoptions.h"
-#include "xcb.h"
 #include "rofi-icon-fetcher.h"
 unsigned int test =0;
 #define TASSERT( a )    {                                 \
-- 
2.33.0

